geest/gui/panels/tree_panel.py

```diff
@@ -905,9 +1001,154 @@ def run_next_worflow_queue(self):
+                "dimensions": self.collect_incomplete_items(root_item, "dimension"),
+                "analysis": self.collect_incomplete_items(root_item, "analysis"),
+            }
+            # Add only non-empty lists to the queue, maintaining the order
+            for role in ["indicators", "factors", "dimensions", "analysis"]:
+                if incomplete_items[role]:
+                    workflow_queue.append(role)
+            QgsMessageLog.logMessage(
+                f"Run All Incomplete Queue: {workflow_queue}",
+                tag="Geest",
+                level=Qgis.Info,
+            )
+
+        elif scenario == "Run Selected":
+            if self.selected_index is not None:
+                workflow_queue = [all_items[self.selected_index]]
+
+        elif scenario == "Run Selected and All Below It":
+            if self.selected_index is not None:
+                workflow_queue = all_items[self.selected_index :]
+
+        elif scenario == "Run Selected and All Incomplete Below It":
+            if self.selected_index is not None:
+                workflow_queue = [
+                    item
+                    for item in all_items[self.selected_index :]
+                    if item.getStatus() != "✔️"
+                ]
+
+        return workflow_queue
+
+    def collect_tree_items(self, root_item):
+        """
+        Recursively collect all items in the tree.
+
+        Args:
+            root_item (JsonTreeItem): The root item of the tree.
+
+        Returns:
+            list: A list of all tree items.
+        """
+        items = []
+
+        def recursive_collect(item):
+            items.append(item)
+            for i in range(item.childCount()):
+                child = item.child(i)
+                recursive_collect(child)
+                QgsMessageLog.logMessage(
+                    f"Item: {child.data(0)} : {child.data(1)}",
+                    tag="Geest",
+                    level=Qgis.Info,
+                )
+
+        QgsMessageLog.logMessage(
+            f"Root Item: {root_item.data(0)} : {root_item.data(1)}",
+            tag="Geest",
+            level=Qgis.Info,
+        )
+
+        recursive_collect(root_item)
+        return items
+
+    def update_selected_index(self, selected, deselected):
+        """
+        Update the selected index based on the current selection in the tree view.
+
+        Args:
+            selected (QItemSelection): The newly selected items.
+            deselected (QItemSelection): The previously selected items.
+        """
+        # Get the first selected index (if multiple items are selected, only the first is considered)
+        indexes = selected.indexes()
+        if indexes:
+            index = indexes[0]  # Take the first selected index
+            self.selected_index = index.row()
+        else:
+            self.selected_index = None
+
+    def collect_incomplete_items(self, parent_item, role):
```

Comment:
This seems like more work than needed. Why not just evaluate each item before creating a workflow and decide if they workflow is needed or not based on the presence of a result_file?